src/beast/beast/http/basic_parser.h

```diff
@@ -90,20 +107,23 @@ class basic_parser
 public:
     using error_code = boost::system::error_code;
 
-    virtual
-    ~basic_parser() = default;
-
     /** Construct the parser.
-        If `request` is `true` this sets up the parser to
-        process an HTTP request.
+
+        @param request If `true`, the parser is setup for a request.
     */
     explicit
-    basic_parser (bool request) noexcept;
+    basic_parser(bool request) noexcept;
 
+    /** Move construct a parser.
+
+        The state of the moved-from object is undefined,
+        but safe to destroy.
+    */
     basic_parser&
-    operator= (basic_parser&& other);
+    operator=(basic_parser&& other);
```

Comment:
I agree, its weird. I intended for the parser to be movable. The `basic_parser` should probably be both movable and copyable, then depending on choice of template argument it might lose those properties.